lib/features/preview-support/PreviewSupport.js

```diff
@@ -175,35 +173,32 @@ PreviewSupport.prototype._cloneMarkers = function(gfx, className = 'djs-dragger'
  * @param {string} [className="djs-dragger"]
  */
 PreviewSupport.prototype._cloneMarker = function(parentGfx, gfx, marker, markerType, className = 'djs-dragger') {
-  var markerId = marker.id + '-' + className;
-
-  var clonedMarker = this._clonedMarkers[ markerId ];
-
-  parentGfx = parentGfx || this._canvas._svg;
 
-  if (!clonedMarker) {
-    clonedMarker = svgClone(marker);
+  // Add a random suffix to the marker ID in case the same marker is previewed multiple times
+  var clonedMarkerId = [ marker.id, className, 'clone', crypto.randomUUID() ].join('-');

```

Comment:
We do not need to persist the Ids, as the preview is ephemeral. I decided to go for `crypto.randomUUID()` instead of the `Ids` library to save on performance + bundle size.
It is supported by all modern browsers
![image](https://github.com/bpmn-io/diagram-js/assets/21984219/a226286e-f70b-4855-ac11-2febeb67715b)

Reply:
How can the same marker be previewed multiple times? I'd rather not depend on too many modern API tools. You could (alternatively) simply generate _one_ static preview ID and add an auto-incrementing number to it.

Reply:
We already have a cheap internal ID generator in place, cf. [`IdGenerator`](https://github.com/bpmn-io/diagram-js/blob/d1caba8b9d4c95f3aa3d53ca58e6bd44ce1c3ccc/lib/util/IdGenerator.js#L11)

Reply:
> How can the same marker be previewed multiple times?

See [7e2bfe3](https://github.com/bpmn-io/diagram-js/pull/909/commits/7e2bfe3bb14dea07998102dcdcb9ba868fd12cb7). When you add the same connection multiple times, the marker will end up with the same copied ID if we do not add randomness.
While it was a bug in this case, I can also see some advanced use-cases where this might be useful (collaborative editing on the same Object?). Let's be robust in the core.